### ISSUE TITLE
Bad errors catch in documentation examples

### ISSUE THREAD

**Issue by cvlab** [nats-io/node-nats#220]
I think many of users, like me, will copy the code from documentation and we expect there to be very good code.

I propose to change from
```javascript
  if(response.code && response.code === NATS.REQ_TIMEOUT) {
    console.log('Request for help timed out.');
    return;
  }
```

to  
```javascript
  if(response instanceof NATS.NatsError && response.code === NATS.REQ_TIMEOUT) {
    console.log('Request for help timed out.');
    return;
  }
```

**Comment by cvlab** [comment]
Do you not see a logic error here, in documentation code? There can be **valid** response object with response.code === 'REQ_TIMEOUT' and it will be wrongly caught

```javascript
// Request for single response with timeout.
nats.requestOne('help', null, {}, 1000, function(response) {
  // `NATS` is the library.
  if(response.code && response.code === NATS.REQ_TIMEOUT) {
    console.log('Request for help timed out.');
    return;
  }
  console.log('Got a response for help: ' + response);
});
```

**Comment by aricart** [comment]
While presumably, you could have configured your client to send and receive JSON messages and in that use case, it is possible that the object would have a `code` property which could be a string and have the actual value of  'REQ_TIMEOUT'. If that is the case, effectively you already know what you are doing.

With that said, currently, I am working on the typescript version of the library which really addresses the issue here, which is the order of the parameters on the callback are suboptimal, because the error parameter should be the first parameter. This would make the client code be simpler as you would always have to test for error `if(err) { ... }` and no parameter shifting is possible.

**Comment by cvlab** [comment]
I agree with `if(err) { ... }`. But now, users need to look inside the code to find `NATS.NatsError`, I hope in documentation to be an example with `NATS.NatsError`. It will save time and there will be an official way to catch errors, that people will use and will be sure that will not change in future minor versions. Thank You!

**Comment by aricart** [comment]
Yes, unfortunately, the current node-nats library mixes optional parameters with non-optional parameters making it difficult to address. Stay tuned!
